qutip/configrc.py

```diff
@@ -56,101 +85,274 @@ def has_qutip_rc():
+    return getattr(config, getter[type(val)])
+
+
+def has_rc_key(key, section=None, rc_file="qutiprc"):
     """
-    Loads the configuration data from the
-    qutiprc file
+    Verify if key exist in section of rc_file
     """
-    config = configparser.ConfigParser()
-    _valid_keys ={'auto_tidyup' : config.getboolean, 'auto_herm' : config.getboolean, 
-            'atol': config.getfloat, 'auto_tidyup_atol' : config.getfloat,
-            'num_cpus' : config.getint, 'debug' : config.getboolean, 
-            'log_handler' : config.getboolean, 'colorblind_safe' : config.getboolean,
-            'openmp_thresh': config.getint}
-    config.read(rc_file)
-    if config.has_section('qutip'):
-        opts = config.options('qutip')
-        for op in opts:
-            if op in _valid_keys.keys():
-                setattr(qset, op, _valid_keys[op]('qutip',op))
-            else:
-                raise Exception('Invalid config variable in qutiprc.')
-    else:
-        raise configparser.NoSectionError('qutip')
-        
-    if config.has_section('compiler'):
-        _valid_keys = ['CC', 'CXX']
-        opts = config.options('compiler')
-        for op in opts:
-            up_op = op.upper()
-            if up_op in _valid_keys:
-                os.environ[up_op] = config.get('compiler', op)
-            else:
-                raise Exception('Invalid config variable in qutiprc.')
- 
- 
-def has_rc_key(rc_file, key):
+    rc_file = full_path(rc_file)
+    if not os.path.isfile(rc_file):
+        return False
     config = configparser.ConfigParser()
     config.read(rc_file)
-    if config.has_section('qutip'):
-        opts = config.options('qutip')
-        if key in opts:
-            return True
-        else:
-            return False
+    if section is None:
+        search_sections = [section for section, _ in sections]
     else:
-        raise configparser.NoSectionError('qutip')
-       
-        
-def write_rc_key(rc_file, key, value):
+        search_sections = [section]
+    for section in search_sections:
+        if config.has_section(section):
+            opts = config.options(section)
+            if key in opts:
+                return True
+    return False
+
+
+def write_rc_key(key, value, section='qutip', rc_file="qutiprc"):
     """
     Writes a single key value to the qutiprc file
-    
+
     Parameters
     ----------
-    rc_file : str
-        String specifying file location.
     key : str
         The key name to be written.
     value : int/float/bool
         Value corresponding to given key.
+    section : str
+        String for which settings object the key belong to.
+        Default : qutip
+    rc_file : str
+        String specifying file location.
+        Default : qutiprc
     """
-    if not os.access(os.path.expanduser("~"), os.W_OK):
+    rc_file = full_path(rc_file)
+    if not os.access(rc_file, os.W_OK):
+        warnings.warn("Does not have permissions to write config file")
```

Comment:
This isn't guaranteed to catch all cases of failed write access - it's probably better just to try it in three lines' time and catch the exception if it fails (see the notes in https://docs.python.org/3/library/os.html#os.access).